fix make cover

#### Expected behavior
make cover to properly generate test coverage html page

#### Actual behavior

$ make cover
# Check for cover :
go test ./ -coverprofile=cover.out -covermode=count; go tool cover -html=cover.out
?       github.com/spacemeshos/go-spacemesh     [no test files]
cover: open cover.out: no such file or directory
make: *** [cover] Error 1


#### Steps to reproduce the actual behavior
Run `make cover` from the project root directory

--- a/.gitignore
+++ b/.gitignore
@@ -17,6 +17,7 @@
 .idea
 .vagrant
 
-go-spacemesh
-cp.out
-cover.out
+/go-spacemesh
+/cp.out
+/cover.out
+/cover-all.out

--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ $(PLATFORMS):
 test:
 	go test ./...
 
-.PHONY: build test
+.PHONY: build test devtools cover
 
 devtools:
 	# Install the build tools
@@ -36,5 +36,8 @@ devtools:
 	govendor sync
 
 cover:
-	# Check for cover :
-	go test ./$(pkg) -coverprofile=cover.out -covermode=count; go tool cover -html=cover.out
+	@echo "mode: count" > cover-all.out
+	@$(foreach pkg,$(PKGS),\
+		go test -coverprofile=cover.out -covermode=count $(pkg);\
+		tail -n +2 cover.out >> cover-all.out;)
+	go tool cover -html=cover-all.out

